tests/phpunit/src/Application/ExceptionApplicationTest.php

```diff
@@ -18,26 +18,49 @@
  */
 class ExceptionApplicationTest extends TestBase {
 
+  protected $kernel;
+
   public function setUp($output = NULL):void {
     parent::setUp($output);
     // If kernel is cached from a previous run, it doesn't get detected in code
     // coverage reports.
     $this->fs->remove('var/cache');
+    $this->kernel = new Kernel('dev', 0);
+    $this->kernel->boot();
+    $this->kernel->getContainer()->set('datastore.cloud', $this->datastoreCloud);
+    $this->kernel->getContainer()->set('Acquia\Cli\Helpers\ClientService', $this->clientServiceProphecy->reveal());
```

Comment:
```
$this->clientServiceProphecy->reveal()
```
The effect of this is basically the same as `getMock()` in the past right, in that it sets a shim-level functional mock of `Acquia\Cli\Helpers\ClientService`?

Reply:
I think that's right. `clientServiceProphecy` is a prophet that lets you mock classes (in this case, `ClientService`), but it's not the mock/double class itself. You have to call `reveal()` to get the mock class.